types/dnstype/messagetypes-string.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package dnstype
+
+import (
+	"errors"
+
+	"golang.org/x/net/dns/dnsmessage"
+)
+
+// ErrUnknownDNSMessageType is returned when an unknown DNS message type string is
+// provided to DNSMessageTypeForString.
+func ErrUnknownDNSMessageType(s string) error {
+	return errors.New("unknown DNS message type: " + s)
+}
```

Comment:
This is a bit strange; typically things with the `Err` prefix are either constants, like [`io/fs.ErrInvalid`](https://pkg.go.dev/io/fs#ErrInvalid), or types; this isn't exactly an error type, so you can't test against it with `errors.Is` like a constant or using `errors.As` like a type. If it doesn't need to be exported, we could just un-export it so it doesn't matter, or if you want it to be useful to compare against, could you make it a type with an `Error() string` method?

Reply:
dropped